expr/func_aggregate.go

```diff
@@ -41,15 +56,34 @@ func (s *FuncAggregate) Exec(dataMap DataMap) ([]models.Series, error) {
 		return series, nil
 	}
 
+	agg := seriesAggregator{function: getCrossSeriesAggFunc(s.name), name: s.name}
+	series = Normalize(dataMap, series)
+	output := aggregate(series, queryPatts, agg, s.xFilesFactor)
+
+	dataMap.Add(Req{}, output)
```

Comment:
if aggregate() is where we retrieve a pointslice from the pool, why don't we make it its responsibility to put it back?
seems weird to do that here.

Reply:
the additional benefit of doing that is right now you have a bug in the `len(series) == 1` case, where `aggregate()` reuses its input, but we add it to the dataMap again, which means the same array would be in the pool twice which leads to nasty bugs.